torch/_refs/__init__.py

```diff
@@ -4966,8 +4966,8 @@ def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLi
             value_ndim == 0,
             lambda: f"only supports a 0-dimensional value tensor, but got tensor with {value_ndim} dimension",
         )
-        # `masked_fill` allows cpu scalar to be moved to cuda but not otherwise.
-        is_cpu_scalar = a.device.type == "cuda" and value.device.type == "cpu"
+        # `masked_fill` allows cpu scalar to be moved to cuda and xpu but not otherwise.
+        is_cpu_scalar = a.device.type in ["cuda", "xpu"] and value.device.type == "cpu"
```

Comment:
Would it be more general to check `a.device.type is not "cpu"`?

Reply:
Allow cpu scalar for torch.where is still not supported in "mps". So I think it can add efforts for mps team to debug if here change to `a.device.type is not "cpu"`